src/cells.cpp

```
@@ -36,14 +43,41 @@ int SmithLife::get_y() { return y; }
+
+
 void SmithLife::compute(Board *board) {
     vector<vector<DiscreteAutomaton*>> grid = board->get_grid();
     for (int i = 0; i < board->get_current_population(); i += 1) {
-        DiscreteAutomaton *cell = grid[i][i];
-        if (cell) {
-            grid[i][i] = new SmithLife(cell->get_x(), cell->get_y(), cell->get_value()+1);
+        for (int j = 0; j < board->get_current_population(); j += 1) {
+            DiscreteAutomaton *cell = dynamic_cast<DiscreteAutomaton*>(board->get_cell(i, j));
+            if (cell) {
+                vector<vector<int>> neighbors = {
```

cppcheck

[performance] Variable 'species' is assigned in constructor body. Consider performing initialization in initialization list.

(useInitializationList)